client/hwlib/src/collectors/os_info.rs

```diff
@@ -68,47 +83,104 @@ impl TryFrom<&Path> for KernelPackage {
     }
 }
 
-pub(crate) fn get_architecture() -> Result<String> {
-    let arch = Command::new("dpkg")
-        .arg("--print-architecture")
-        .output()?
-        .stdout;
-    Ok(String::from_utf8(arch)?.trim().to_string())
+pub(crate) fn get_architecture(runner: &impl CommandRunner) -> Result<String> {
+    let arch = runner.run_command("dpkg", &["--print-architecture"])?;
+    Ok(arch.trim().to_owned())
 }
 
-pub(super) fn get_codename() -> Result<String> {
+pub(super) fn get_codename(runner: &impl CommandRunner) -> Result<String> {
     lazy_static! {
         static ref CODENAME_RE: Regex = Regex::new(r"Codename:\s*(\S+)").unwrap();
     }
-    get_lsb_release_info("-c", &CODENAME_RE)
+    get_lsb_release_info("-c", &CODENAME_RE, runner)
 }
 
-pub(super) fn get_distributor() -> Result<String> {
+pub(super) fn get_distributor(runner: &impl CommandRunner) -> Result<String> {
     lazy_static! {
         static ref DISTRIBUTOR_RE: Regex = Regex::new(r"Distributor ID:\s*(\S+)").unwrap();
     }
-    get_lsb_release_info("-i", &DISTRIBUTOR_RE)
+    get_lsb_release_info("-i", &DISTRIBUTOR_RE, runner)
 }
 
-pub(super) fn get_version() -> Result<String> {
+pub(super) fn get_version(runner: &impl CommandRunner) -> Result<String> {
     lazy_static! {
         static ref VERSION_RE: Regex = Regex::new(r"Release:\s*(\S+)").unwrap();
     }
-    get_lsb_release_info("-r", &VERSION_RE)
+    get_lsb_release_info("-r", &VERSION_RE, runner)
 }
 
-fn get_lsb_release_info(flag: &str, re: &Regex) -> Result<String> {
-    let lsb_release_output = Command::new("lsb_release")
-        .arg(flag)
-        .output()
-        .context(format!(
-            "failed to execute lsb_release command with flag {}",
-            flag
-        ))?;
-    let output_str = String::from_utf8(lsb_release_output.stdout)
-        .context("failed to convert lsb_release output to UTF-8 string")?;
-    re.captures(&output_str)
+fn get_lsb_release_info(flag: &str, re: &Regex, runner: &impl CommandRunner) -> Result<String> {
+    let lsb_release_output = runner.run_command("lsb_release", &[flag])?;
+    re.captures(&lsb_release_output)
         .and_then(|caps| caps.get(1))
         .map(|m| m.as_str().to_string())
         .ok_or_else(|| anyhow!("failed to capture information using regex"))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::helpers::test_utils::get_test_filepath;
+    use std::collections::HashMap;
+
+    // Type aliases for better readability
+    type CommandKey<'args> = (&'args str, Vec<&'args str>);
```

Comment:
I think putting 'key' in the name of this type feels slightly off as it leaks an implementation detail---the current API specifically hides the use of the map behind the scenes. I think instead, we should focus on what this represents rather than how it's used; as we're mocking the `SystemCommandRunner`, how about calling this type `SystemCommand`?